pkgs/top-level/python-packages.nix

```diff
@@ -5184,6 +5184,24 @@ in {
     };
   };
 
+  pytest-datafiles = buildPythonPackage rec {
+    pname = "pytest-datafiles";
+    name = "${pname}-${version}";
+    version = "1.0";
+    src = fetchPypi {
+      inherit version pname;
+      sha256 = "1w5435b5pimk6479ml53lmld3qbag7awcg4gl3ljdywc1v096r5v";
+    };
+
+    propagatedBuildInputs = with self; [ py pytest ];
```

Comment:
do not propagate `pytest`, instead add it only as a `buildInput`. `pytest` is a bit special and should never be propagated.